x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/rules/50_query_rule_put.yml

```diff
@@ -0,0 +1,114 @@
+setup:
+  - requires:
+      cluster_features: [ "gte_v8.15.0" ]
+      reason: Introduced in 8.15.0
+
+---
+'Create query rule with existing ruleset':
+  - do:
+      query_ruleset.put:
+        ruleset_id: test-ruleset
+        body:
+          rules:
+            - rule_id: query-rule-id1
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  values: [ elastic ]
+              actions:
+                ids:
+                  - 'id1'
+                  - 'id2'
+            - rule_id: query-rule-id2
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  values: [ kibana ]
+              actions:
+                docs:
+                  - '_index': 'test-index1'
+                    '_id': 'id3'
+                  - '_index': 'test-index2'
+                    '_id': 'id4'
+              priority: 1
+
+  - match: { result: 'created' }
+
+  - do:
+      query_rule.put:
+        ruleset_id: test-ruleset
+        rule_id: query-rule-id3
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'logstash' ]
+          actions:
+            ids:
+              - 'id1'
+          priority: 2
+
+  - match: { result: 'updated' }
+
+---
+'Create Query Rule - Resource already exists':
+  - do:
+      query_rule.put:
+        ruleset_id: test-query-rule-recreating
+        rule_id: abc
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'elastic' ]
+          actions:
+            ids:
+              - 'id1'
+          priority: 5
+
+  - match: { result: 'created' }
+
+  - do:
+      query_rule.put:
+        ruleset_id: test-query-rule-recreating
+        rule_id: abc
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'elastic' ]
+          actions:
+            ids:
+              - 'id2'
+          priority: 3
+
+  - match: { result: 'updated' }
+
+---
+'Create Query Rule - Insufficient privilege':
+  - skip:
+      features: headers
+
+  - do:
+      catch: forbidden
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      query_rule.put:
+        ruleset_id: forbidden-query-ruleset
+        rule_id: abc
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'elastic' ]
+          actions:
+            ids:
+              - 'id1'
+              - 'id2'
+
+  - match: { error.type: 'security_exception' }
```

Comment:
unrelated to your PR - but it looks like the yml tests for query ruleset deletion and listing could benefit from a similar test to check for security privileges.